src/avatar.scss

```diff
@@ -0,0 +1,153 @@
+@import "./mixins";
+
+/*!
+.fd-avatar+((--circle), (--thumbnail), (--xxs|--xs|--s|--m|--l|--xl|--xxl))
+*/
+$block: #{$fd-namespace}-avatar;
+
+.#{$block} {
+  // LOCAL VARS (set all themeable properties, always include !default)
+  $fd-const-ratio: 1rem;
+  $fd-avatar-sizes: (
+    xs: $fd-const-ratio * 2,
+    s: $fd-const-ratio * 3,
+    m: $fd-const-ratio * 4,
+    l: $fd-const-ratio * 5,
+    xl: $fd-const-ratio * 7,
+  ) !default;
+  $fd-avatar-border-radius: $fd-border-radius !default;
+  $fd-avatar-border-fade: rgba(0, 0, 0, 0.5); // Should be fade(var(--sapContent_ShadowColor), 10);
+  $fd-avatar-color: var(--sapContent_ContrastTextColor) !default;
+  $fd-avatar-background-color: var(--sapAccentColor6) !default;
+
+  // BLOCK BASE *******************************************
+  @at-root {
+    [class*="#{$block}"] {
+      @include fd-reset();
+
+      color: $fd-avatar-color;
+      background-color: $fd-avatar-background-color;
+      display: inline-block;
+      position: relative;
+      vertical-align: middle;
+      text-align: center;
+      background-repeat: no-repeat;
+      background-size: cover;
+      background-position: 50%;
+      border-radius: $fd-avatar-border-radius;
+    }
+  }
+  // BLOCK MODIFIERS ************
+  // sizes
+  @each $size, $value in $fd-avatar-sizes {
+    &.#{$block}--#{$size} {
+      @include fd-reset-spacing();
+
+      width: $value;
+      height: $value;
+      max-width: $value;
+      max-height: $value;
+      min-width: $value;
+      min-height: $value;
+      font-weight: 400;
+
+      @if $size == xs or $size == s or $size == m {
```

Comment:
Instead of @if blocks you can use the approach we have Info Label for example where for each size you can have value, line-heigh, font size and then with one block of code set these values. 
Example from Info label:
```
  $color-accents: (
    "1": ("background": var(--sapLegendBackgroundColor1), "border": var(--sapLegendColor1)),
    "2": ("background": var(--sapLegendBackgroundColor2), "border": var(--sapLegendColor2)),
    "3": ("background": var(--sapLegendBackgroundColor3), "border": var(--sapLegendColor3))
  );

@each $set-name, $color-set in $color-accents {
    &--accent-color-#{$set-name} {
      background-color: map_get($color-set, "background");
      border-color: map_get($color-set, "border");
    }
  }
```